src/server/active_tran_server.cpp

```diff
@@ -164,8 +161,30 @@ active_tran_server::push_request (ats_to_ps_request reqid, std::string &&payload
   m_ps_request_queue->push (reqid, std::move (payload));
 }
 
-void
-active_tran_server::receive_saved_lsa (cubpacking::unpacker &upk)
+void active_tran_server::receive_log_page (cubpacking::unpacker &upk)
+{
+  std::string message;
+  upk.unpack_string (message);
+
+  int error_code;
+  memcpy (&error_code, message.c_str (), sizeof (error_code));
+
+  LOG_PAGEID pageid = 0;
+  if (error_code == NO_ERROR)

```

Comment:
Also add a message for error_code != NO_ERROR case, to help troubleshooting when it occurs.

```
  if (error_code == NO_ERROR)
    {
      // read log page
      // add message with log pageid
    }
  else
    {
      // add message with error code
    }
```